build-system/server/new-server/transforms/sxg/sxg-transform.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PostHTML} from 'posthtml';
+import {URL} from 'url';
+import {isValidScript, ScriptNode} from '../utilities/script';
+import {CDNURLToLocalDistURL} from '../utilities/cdn';
+import {OptionSet} from '../utilities/option-set';
+import minimist from 'minimist';
+const argv = minimist(process.argv.slice(2));
+
+/**
+ * @param head
+ * @param script
+ * @param compiled 
+ */
+function appendSxgScript(head: PostHTML.Node, script: ScriptNode, compiled: boolean): void {
+  if (argv.compiled || compiled) {
+    let sxgPath = CDNURLToLocalDistURL(
+      new URL(script.attrs.src || ''),
+      undefined
+    ).toString();
+    script.attrs.src = sxgPath.replace('.js', '.sxg.js');
+  }
+  else {
+    const urlName = script.attrs.src.toString();
+    if (urlName.includes('js?')){
+        script.attrs.src = urlName + '&f=sxg';
```

Comment:
I'm a little concerned on how line may be handled.

Reply:
how so?

Reply:
construct an actual URL object and then use https://nodejs.org/api/url.html#url_url_searchparams to append

Reply:
Done.